example/widget_surveyor.dart

```diff
@@ -183,6 +187,28 @@ class AnalysisResults with IterableMixin<AnalysisResult> {
       };
 }
 
+// bug: fixed in linter 0.1.116 (remove once landed)
+// ignore: prefer_mixin
+class CSVResultWriter {
+  final AnalysisResults results;
+  CSVResultWriter(this.results);
+
+  void write() {
+    var buffer = StringBuffer();
+    for (var result in results) {
+      for (var entry in result.widgetReferences.entries) {
+        var references = entry.value;
+        var widgetId = entry.key.replaceAll('#', ',');
+        for (var ref in references) {
+          buffer.writeln('$widgetId,$ref');
+        }
+      }
+    }
+
+    File('results.csv').writeAsStringSync(buffer.toString());
```

Comment:
It would be more efficient to create a sink on the file (I'm thinking `openWrite`) and just write to it. That way we don't have to store the entire file content in memory, which might become a problem with larger corpuses.

Also, at some future date, consider adding a command-line option to enable CSV output to a particular file (such as `--csv=results.csv`) in order to support multiple output formats and customizing the location of the output artifacts.

Reply:
Agreed on both counts and on my punch list.  Thanks!